### PR TITLE
fix: promises cancellation leaving invalid state

### PR DESCRIPTION
**Pull request by BrianAmadori**
When any asset bundle promise is cancelled while its loading, there's was the chance of:

- AssetBundle being kept in memory partially when it shouldn't (leading to the "AssetBundle already in memory" errors)
- AssetBundle depmap download being not cancelled properly, leaving invalid state (i.e. waiting to be downloaded by other assets ad eternum)
- Dependent assets being not cancelled properly.
- Other issues i'm not aware of.

This may be related to the invisible or invalid wearables issue. Because it was detected that unloading/loading them again was leading to invalid state, pink wearables, nullRef errors etc.

### CODE REVIEW

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/NFTItemToggle.cs [hunk 1]
```diff
@@ -18,9 +18,9 @@ protected override void Awake()
         OnHideAllInfos += HideInfo;
 
         HideInfo();
-        infoButton.onClick.AddListener(ToggleInfo);
-        closeInfoButton.onClick.AddListener(HideInfo);
-        sellButton.onClick.AddListener(CallOnSellClicked);
+        infoButton?.onClick?.AddListener(ToggleInfo);
```
Comment (AjimenezDCL) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/NFTItemToggle.cs]: Why is this needed? Those items should be provided by unity's serialization and impossible to be null

**Review by AjimenezDCL — APPROVED** [review]

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/AssetBundles/AB/AssetPromise_AB.cs [hunk 2]
```diff
@@ -87,16 +103,17 @@ protected override void OnBeforeLoadOrReuse()
 
         protected IEnumerator LoadAssetBundleWithDeps(string baseUrl, string hash, Action OnSuccess, Action OnFail)
         {
-            yield return DependencyMapLoadHelper.GetDepMap(baseUrl, hash);
+            if (!DependencyMapLoadHelper.dependenciesMap.ContainsKey(hash))
+                CoroutineStarter.Start(DependencyMapLoadHelper.GetDepMap(baseUrl, hash));
 
-            if (DependencyMapLoadHelper.dependenciesMap.ContainsKey(hash))
+            yield return DependencyMapLoadHelper.WaitUntilDepMapIsResolved(hash);
```
Comment (pravusjif) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/AssetBundles/AB/AssetPromise_AB.cs]: I checked the failing tests (`Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/AssetBundles/AB/Tests/BlockedAndMasterPromisesShould.cs` -> `FailCorrectlyWhenGivenWrongURL()` && `Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/AssetBundles/AB_GameObject/Tests/BlockedAndMasterPromisesShould.cs` -> `FailCorrectlyWhenGivenWrongURL()`) and I'm not sure if that ` yield return DependencyMapLoadHelper.WaitUntilDepMapIsResolved(hash)` will continue waiting forever when a wrong URL has been used?

**Review by pravusjif — APPROVED** [review]

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/AssetBundles/AB/AssetPromise_AB.cs (hunk 2 above)
Comment (BrianAmadori) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/AssetBundles/AB/AssetPromise_AB.cs]: Yep, this may happen. Checking it out.

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/NFTItemToggle.cs (hunk 1 above)
Comment (BrianAmadori) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/NFTItemToggle.cs]: This was making tests fail by some sorcery. Maybe it was Unity library having a weird state. Going to remove it and check if this still happens.
